### PR TITLE
Add tasks update action to rule engine

### PR DESCRIPTION
**Pull request by Badoche1337**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

This PR add a new specific action in rule action, this action allow to update tasks state of a ticket in rule engine

### CODE REVIEW

File: inc/ruleticket.class.php [hunk 1]
```diff
@@ -206,6 +206,17 @@ function executeActions($output, $params) {
                      }
                   }
 
+                  // special case of tasks status
+                  if ($action->fields["field"] == '_tasks_status') {
+                     $tickettask = new TicketTask();
+                     $tasks = $tickettask->find('state IN (0, 1) AND tickets_id = '.$output['id']);
```
Comment (trasher) [inline inc/ruleticket.class.php]: Please avoid using raw SQL

File: inc/tickettask.class.php [hunk 2]
```diff
@@ -251,6 +251,48 @@ function getRights($interface = 'central') {
       return $values;
    }
 
+   /**
+    * Dropdown of tickettask state
+    *
+    * @param $name   select name
```
Comment (trasher) [inline inc/tickettask.class.php]: missing type (syntax is `@param type $varname Documentation`)
also missing `@return`

Comment (trasher) [reply inc/tickettask.class.php]: (same for other docblocks)

File: inc/ruleticket.class.php [hunk 3]
```diff
@@ -661,6 +676,11 @@ function getActions() {
       $actions['requesttypes_id']['type']                   = 'dropdown';
       $actions['requesttypes_id']['table']                  = 'glpi_requesttypes';
 
+      $actions['_tasks_status']['name']                     = _n('Ticket task', 'Ticket tasks', 2).' - '.__('Statut');
```
Comment (cedric-anne) [inline inc/ruleticket.class.php]: Typo: `Statut` -> `Status`.

File: inc/tickettask.class.php [hunk 4]
```diff
@@ -251,6 +251,52 @@ function getRights($interface = 'central') {
...
+   /**
+    * Dropdown of tickettask state
+    *
+    * @param string $name   select name
+    * @param string $value  default value (default '')
+    * @param boolean $display  display of send string ? (true by default)
+    * @param array $options  options
+    *
+    * @return string from Dropdown::showFromArray function (html output)
+   **/
```
Comment (cedric-anne) [inline inc/tickettask.class.php]: Missing `@since 9.4.0` tag.

File: inc/tickettask.class.php [hunk 5]
```diff
@@ -251,6 +251,52 @@ function getRights($interface = 'central') {
...
+                                                                  'display' => $display], $options));
+   }
+
+    /**
+    * Get object state name
+    *
+    * @param integer $value state ID
+    *
+    * @return string translation for value supplied
+   **/
```
Comment (cedric-anne) [inline inc/tickettask.class.php]: Missing `@since 9.4.0` tag.

**Comment by trasher** [comment]
@Badoche1337 ping? This one is stalled from a while; I'll close iff we do not have news (alos, you have to rebase to fix the conflict).

**Pull request closed without merging**